proxy/http/remap/NextHopHealthStatus.cc

```diff
@@ -76,6 +76,7 @@ NextHopHealthStatus::markNextHop(TSHttpTxn txn, const char *hostname, const int
   int64_t fail_threshold  = sm->t_state.txn_conf->parent_fail_threshold;
   int64_t retry_time      = sm->t_state.txn_conf->parent_retry_time;
   uint32_t new_fail_count = 0;
+  const char *host        = sm->t_state.request_data.get_host();
```

Comment:
Should we add `ink_assert(host != nullptr);` here?

Reply:
If I understand asserts in c++ properly, this would essentially kill ATS if the host is null, right? Do we really want to kill the entire ATS program if there is a single malformed URL host being sent?

Reply:
We could check for nullptr and write a more verbose message if host is not null. If it is, we can write a generic response.

Reply:
See [9ce7bdd](https://github.com/apache/trafficserver/pull/9960/commits/9ce7bdda709a5107b2c000f7c63451b531c5b3ae) for what I am talking about.

Reply:
I guess it's possible/likely that the printf-like debug output call checks for and handles nullptr.

Yes, I think it's best that, when we make wrong assumptions about code invariants, that the application exit in a way that makes it clear what our wrong assumption is.  Rather than seeing a crash, and not being sure what the proximate cause was.

Reply:
Remember that ink_assert() checks are only done in debug builds.  It would not affect release builds.